src/ast/ast.h

```diff
@@ -533,6 +537,46 @@ class Variable : public Node {
   SizedType var_type;
 };
 
+class VariableAddr : public Node {
+public:
+  explicit VariableAddr(ASTContext &ctx, Variable *var, Location &&loc)
+      : Node(ctx, std::move(loc)),
+        var(var),
+        var_addr_type(CreatePointer(CreateNone())) {};
+  explicit VariableAddr(ASTContext &ctx,
+                        const VariableAddr &other,
+                        const Location &loc)
+      : Node(ctx, loc + other.loc),
+        var(other.var),
+        var_addr_type(other.var_addr_type) {};
+
+  const SizedType &type() const
+  {
+    assert(var_addr_type.IsPtrTy());
```

Comment:
This is not valid; it may not be resolved yet. This type will be None until the underlying variable type is resolved.

Reply:
Ack.